pkg/cfn/builder/api_test.go

```diff
@@ -0,0 +1,229 @@
+package builder_test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/weaveworks/eksctl/pkg/cfn/builder"
+	"github.com/weaveworks/eksctl/pkg/cloudconfig"
+	"github.com/weaveworks/eksctl/pkg/eks/api"
+)
+
+type Template struct {
+	Resources map[string]struct {
+		Properties struct {
+			Tags []struct {
+				Key   interface{}
+				Value interface{}
+
+				PropagateAtLaunch string
+			}
+			UserData string
+		}
+	}
+}
+
+func newStackWithOutputs(outputs map[string]string) cfn.Stack {
+	s := cfn.Stack{}
+	for k, v := range outputs {
+		func(k, v string) {
+			s.Outputs = append(s.Outputs,
+				&cfn.Output{
+					OutputKey:   &k,
+					OutputValue: &v,
+				})
+		}(k, v)
+	}
+	return s
+}
+
+var _ = Describe("CloudFormation template builder API", func() {
```

Comment:
For now I've written test just for the most complicated and deep parts of this, we can probably add a few broader tests that – not sure yet what those should be.